src/ppopt/utils/region_overlap_utils.py

```diff
@@ -0,0 +1,188 @@
+from itertools import permutations
```

Comment:
Can you add docstrings to all functions in this file giving a short description of what they are doing using the reST format